File: app/src/main/java/com/example/retrofitgithubapi/activities/java/LogInActivity.java
package com.example.retrofitgithubapi.activities.java;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.example.retrofitgithubapi.R;
import com.google.android.material.textfield.TextInputLayout;

public class LogInActivity extends AppCompatActivity {

    private TextInputLayout mInputUserName;
    private Button mLoginBtn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        mInputUserName = findViewById(R.id.input_username);
        mLoginBtn = findViewById(R.id.login_btn);

        mLoginBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                String inputUserName = mInputUserName.getEditText().getText().toString().trim();

                if (inputUserName.isEmpty()) {
                    mInputUserName.setError("Enter UserName");
                } else {
                    mInputUserName.setErrorEnabled(false);
                    startActivity(new Intent(getApplicationContext(), UserActivity.class).putExtra("STRING_I_NEED", inputUserName));
                }


            }
        });

    }
}

File: app/src/main/java/com/example/retrofitgithubapi/activities/kotlin/LoginActivity.kt
package com.example.retrofitgithubapi.activities.kotlin

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.example.retrofitgithubapi.R
import com.example.retrofitgithubapi.activities.java.UserActivity
import com.google.android.material.textfield.TextInputLayout
import kotlinx.android.synthetic.main.activity_login.*

class LoginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        login_btn.setOnClickListener {
            val inputUserName: String = input_username.editText!!.text.toString().trim { it <= ' ' }
            if (inputUserName.isEmpty()) {
                input_username.error = "Enter UserName"
            } else {
                input_username.isErrorEnabled = false
                startActivity(Intent(applicationContext, UserActivityKt::class.java).putExtra("STRING_I_NEED", inputUserName))
            }
        }

    }
}